seq2science/rules/configuration_generic.smk

```diff
@@ -114,23 +114,24 @@ if len(errors):
     logger.error("")  # empty line
     raise TerminatedException
 
-# for each column, if found in samples.tsv:
+# for each of these functional columns, if found in samples.tsv:
 # 1) if it is incomplete, fill the blanks with replicate/sample names
-# (sample names if replicates are not found/applicable)
-# 2) drop column if it is identical to the replicate/sample column, or if not needed
+#    (sample names if replicates are not found/applicable)
+# 2) drop column if it provides no information
+#    (renamed in case it's used in a DE contrast)
 if 'technical_replicate' in samples:
     samples['technical_replicate'] = samples['technical_replicate'].mask(pd.isnull, samples['sample'])
     if len(samples['technical_replicate'].unique()) == len(samples['sample'].unique()) or config.get('technical_replicates') == 'keep':
-        samples = samples.drop(columns=['technical_replicate'])
+        samples.rename(columns={'technical_replicate': '_trep'}, inplace=True)
```

Comment:
why are we renaming to these names?

Reply:
testing showed me an edge case:
- your DE contrast is `biological_replicate_group2_group1`
- your config option `biological_replicates` is set to `keep`
- s2s would then drop this column internally (less keyword columns = simpler logic, so I'd keep it this way)

But now your DE contrast column is gone. So the DE contrast check fails.
To fix it I rename the column instead of dropping it, and we check these for DE contrasts too ([here](https://github.com/vanheeringen-lab/seq2science/blob/a8964977cb5d9d18b6f3a22cb4eebe1fff8567c3/seq2science/util.py#L248)).

In DESeq2.R, we reload the samples.tsv file, so the `biological_replicate` column is still there. Because the check was already performed in s2s, we know the contrast is valid.